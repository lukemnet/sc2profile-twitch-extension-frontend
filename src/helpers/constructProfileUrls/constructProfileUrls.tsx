import StarCraft2API, { PlayerObject }  from 'starcraft2-api';

const constructProfileUrls = (profiles: PlayerObject[]) =>
  profiles.map((profile: PlayerObject) =>
    StarCraft2API.constructProfileUrl({ ...profile }, profile.locale));

export default constructProfileUrls;
