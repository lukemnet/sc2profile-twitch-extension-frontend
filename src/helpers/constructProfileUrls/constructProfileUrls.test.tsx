import constructProfileUrls from './constructProfileUrls';

const profiles = [
  {
    regionId: 1,
    realmId: 1,
    profileId: '123',
    locale: 'en-us',
  },
  {
    regionId: 2,
    realmId: 2,
    profileId: '456',
    locale: 'ru-ru',
  },
  {
    regionId: 3,
    realmId: 1,
    profileId: '789',
    locale: 'pl-pl',
  }
];

it('constructs valid SC2 profiles', () => {
  expect(constructProfileUrls(profiles)).toMatchSnapshot();
});