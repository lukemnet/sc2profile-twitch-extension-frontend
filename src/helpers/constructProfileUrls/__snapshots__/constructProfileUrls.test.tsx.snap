// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constructs valid SC2 profiles 1`] = `
Array [
  "https://starcraft2.com/en-us/profile/1/1/123",
  "https://starcraft2.com/ru-ru/profile/2/2/456",
  "https://starcraft2.com/pl-pl/profile/3/1/789",
]
`;
